rabbitmq/src/main/java/io/micronaut/rabbitmq/bind/RabbitMessageCloseable.java

```diff
@@ -79,11 +80,36 @@ public void close() throws MessageAcknowledgementException {
      * Set to false if the message should be rejected.
      * Set to null if the message should not be acknowledged or rejected.
      *
+     * @deprecated as of 3.4.0
+     *
      * @param acknowledge The acknowledge parameter.
      * @return The same instance
      */
+    @Deprecated
     public RabbitMessageCloseable withAcknowledge(@Nullable Boolean acknowledge) {
-        this.acknowledge = acknowledge;
+        if (acknowledge == null) {
+            acknowledgmentAction = AcknowledgmentAction.NONE;
+        } else if (Boolean.TRUE.equals(acknowledge)) {
+            acknowledgmentAction = AcknowledgmentAction.ACK;
+        } else {
+            acknowledgmentAction = AcknowledgmentAction.NACK;
+        }
+        return this;
+    }
+
+    /**
+     * Builder style sets whether the message acknowledgment action should be addressed and message should be acknowledged or rejected,
+     * or otherwise should acknowledgment action be skipped.
+     *
+     * Set to {@link AcknowledgmentAction#ACK} if the message should be acknowledged.
+     * Set to {@link AcknowledgmentAction#NACK} if the message should be rejected.
+     * Set to {@link AcknowledgmentAction#NONE} if the message should not be acknowledged or rejected.
+     *
+     * @param acknowledgmentAction The acknowledgment action parameter.
+     * @return The same instance
+     */
+    public RabbitMessageCloseable withAcknowledgmentAction(AcknowledgmentAction acknowledgmentAction) {
```

Comment:
add `@since` and whatever the next version will be and add nullability annottations